proto/cosmos/crypto/v1/keys.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+package cosmos.crypto.v1;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/any.proto";
+
+option go_package = "github.com/cosmos/cosmos-sdk/crypto/keys";
+
+// LegacyAminoMultisigThresholdPubKey specifies a public key type 
+// which nests multiple public keys and a threshold,
+// it uses legacy amino address rules
+message LegacyAminoMultisigThresholdPubKey {
+    option (gogoproto.goproto_getters) = false;
+
+    uint32 threshold = 1 [(gogoproto.customname) = "K", (gogoproto.moretags) = "yaml:\"threshold\""];
+    repeated google.protobuf.Any public_keys = 2 [(gogoproto.customname) = "PubKeys", (gogoproto.moretags) = "yaml:\"pubkeys\""];
+}
+
+// Secp256K1PubKey defines a secp256k1 public key
+message Secp256K1PubKey {
+    bytes key = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1.PubKey"];
+}
+
+// Secp256K1PrivKey defines a secp256k1 private key
+message Secp256K1PrivKey {
+    bytes key = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1.PrivKey"];
```

Comment:
Another idea would be to remove casttype, remove the `secp256k1.PubKey` & `secp256k1.PrivKey` types altogether, and implement the PubKey interface directly on the proto types (so that we don't have 2 types anymore).

I don't know which way is cleaner, I'm also fine with the current approach.

Reply:
Having less (duplicate) types is ideal IMHO, especially if they're semantically equivalent.

Reply:
It sounds preferable indeed, I'll work on that.